### PR TITLE
Feat/k8s no clustername

### PR DESCRIPTION
**Pull request by nfuden**
This pr is split from https://github.com/solo-io/skv2/pull/393

This PR should not update go mod and purely enable preexisting clustername usages to instead work via the annotation storage method.
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise/issues/5158

### CODE REVIEW

File: pkg/resource/mocks/client.go [hunk 1]
```diff
@@ -38,17 +38,22 @@ func (m *MockClient) EXPECT() *MockClientMockRecorder {
 }
 
 // Get mocks base method
-func (m *MockClient) Get(ctx context.Context, key client.ObjectKey, obj client.Object) error {
+func (m *MockClient) Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
```
Comment (jenshu) [inline pkg/resource/mocks/client.go:41]: also i think this can be reverted

File: crds/multicluster.solo.io_v1alpha1_crds.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    crd.solo.io/specHash: ad9e1dd2ecf8891d
+    crd.solo.io/specHash: 396fd84315636ecb
```
Comment (jenshu) [inline crds/multicluster.solo.io_v1alpha1_crds.yaml:7]: was this change auto-generated?

File: pkg/multicluster/register/registrant.go [hunk 3]
```diff
@@ -594,26 +594,64 @@ func (c *clusterRegistrant) getTokenForSa(
...
+
+	// create a secret name format. This is similar to what was previously done
+	// but after the k8s upgrade we no longer have knowledge of what was previously
+	// autogenerated by k8s.
+	// TODO(nfuden): move away from these long lived connections as they are
+	// likely to be deprecated even more in the future.
+	secretName := fmt.Sprintf("%s-secret", sa.Name)
+
+	// see if there is a secret
+	_, err = remoteSecretClient.GetSecret(ctx, client.ObjectKey{Name: secretName, Namespace: saRef.Namespace})
```
Comment (jenshu) [inline pkg/multicluster/register/registrant.go:618]: this should be moved into the retry loop, and why is there a separate call to GetSecret in the retry below?

Comment (nfuden) [reply pkg/multicluster/register/registrant.go:618]: At this point we do not believe there exists a secret already. If there is we should not destroy it by our upsert.
This is why we dont have this in the loop.

The loop get secret is to make sure the upsert and k8s controller update has taken place

**Review by jenshu — APPROVED** [review]

**Review by conradhanson — APPROVED** [review]

**Pull request merged**